tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -57,40 +83,24 @@ Status XlaGpuDeviceFactory::CreateDevices(
   }
   string allowed_gpus =
       session_options.config.gpu_options().visible_device_list();
-  std::set<int> gpu_ids;
+  absl::optional<std::set<int>> gpu_ids = ParseVisibleDeviceList(allowed_gpus).ValueOrDie();
+  // We want to fill the gpu_ids set with all devices if config string is empty.
   int num_visible_devices = platform.ValueOrDie()->VisibleDeviceCount();
```

Comment:
Move this line into the `if` statement?  Or even just change the loop to

```
for (int i = 0; i < platform.ValueOrDie()->VisibleDeviceCount(); ++i)
```

This has the advantage that now the comment on the line above is right above the if statement.  Makes a lot more sense.

Reply:
I forgot to update the comment when I made the earlier change. Sorry about that. It is fixed now.